pkg/rke2/rke2.go

```diff
@@ -57,23 +54,22 @@ func Agent(clx *cli.Context, cfg Config) error {
 
 func setup(clx *cli.Context, cfg Config) error {
 	var dataDir string
+
 	for _, f := range clx.Command.Flags {
 		switch t := f.(type) {
 		case cli.StringFlag:
 			if strings.Contains(t.Name, "data-dir") {
-				dataDir = t.Value
+				dataDir = *t.Destination
```

Comment:
Value holds the default value, the actual user-provided input is stored in Destination.